moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -443,11 +452,14 @@ std::string MoveItConfigData::getGazeboCompatibleURDF()
       }
       else if (static_cast<std::string>(doc_element->Value()).find("joint") != std::string::npos)
       {
-        // Before adding a transmission element, make sure there the joint is not fixed
-        if (static_cast<std::string>(doc_element->Attribute("type")) != "fixed")
+        // Before adding a transmission element, make sure there is none and the joint is not fixed
+        std::string joint_name = static_cast<std::string>(doc_element->Attribute("name"));
+        if (static_cast<std::string>(doc_element->Attribute("type")) != "fixed" &&
+            !std::any_of(trans_elements.begin(), trans_elements.end(), [joint_name](const std::string& trans) {
```

Comment:
Has to be improved doesn't work when the name is different.

Reply:
I improved upon this by using https://github.com/ros-planning/moveit/pull/2960/commits/74f7bd3f4c7ad60af5f7ac8f62ee85743ff007ae.